gcp/api/server.py

```diff
@@ -1179,6 +1221,26 @@ def is_cloud_run() -> bool:
   return os.getenv('K_SERVICE') is not None
 
 
+def _is_affected(ecosystem: Ecosystem, version: str, affected_package: osv.AffectedPackage) -> bool:
+  """Checks if a version is affected."""
+  for range in affected_package.ranges:
+    range: osv.AffectedRange2
+
+    introduced_version = '0'
+    fixed_version = '99999'
+    for event in range.events:
+      event: osv.AffectedEvent
+      if event.type == 'introduced':
+        introduced_version = event.value
+      elif event.type == 'fixed':
+        fixed_version = event.value
+
+    if ecosystem.sort_key(version) >= ecosystem.sort_key(introduced_version) and ecosystem.sort_key(version) < ecosystem.sort_key(fixed_version):
```

Comment:
A few things:
- You should use `None` (or similar) to represent a missing fixed version. `99999` might not parse in some ecosystems, or it could be smaller than a legitimate version e.g. CalVer like `20240704`
- You should also be checking for `last_affected` events.
- There can be multiple `introduced`/`fixed`/`last_affected` events in an affected range (and they're not guaranteed to be sorted)

You could probably just reuse the logic in `_is_semver_affected()`:
https://github.com/google/osv.dev/blob/6084fc7849a1e32dae4a9a09fd56e4b4417377d8/gcp/api/server.py#L862-L876